rustls-platform-verifier/src/lib.rs

```diff
@@ -97,16 +52,22 @@ pub trait BuilderVerifierExt {
     /// use rustls_platform_verifier::BuilderVerifierExt;
     /// let config = ClientConfig::builder()
     ///     .with_platform_verifier()
+    ///     .unwrap()
     ///     .with_no_client_auth();
     /// ```
-    fn with_platform_verifier(self) -> ConfigBuilder<ClientConfig, WantsClientCert>;
+    fn with_platform_verifier(
+        self,
+    ) -> Result<ConfigBuilder<ClientConfig, WantsClientCert>, rustls::Error>;
 }
 
 impl BuilderVerifierExt for ConfigBuilder<ClientConfig, WantsVerifier> {
-    fn with_platform_verifier(self) -> ConfigBuilder<ClientConfig, WantsClientCert> {
+    fn with_platform_verifier(
+        self,
+    ) -> Result<ConfigBuilder<ClientConfig, WantsClientCert>, rustls::Error> {
         let provider = self.crypto_provider().clone();
-        self.dangerous()
-            .with_custom_certificate_verifier(Arc::new(Verifier::new().with_provider(provider)))
+        Ok(self
+            .dangerous()
+            .with_custom_certificate_verifier(Arc::new(Verifier::new(provider)?)))
```

Comment:
Very small nit: Now that we have a `?` it reads clearer if the fallible part is extracted out of the parentheses soup:
```rust
let provider = self.crypto_provider().clone();
let verifier = Verifier::new(provider)?;
Ok(self.dangerous().with_custom_certificate_verifier(Arc::new(verifier)))
```

Reply:
Rearranged this slightly, although I still inlined the `provider` into the `verifier` setup statement.

Reply:
That looks much better as well, thanks!